clang/include/clang/Basic/AttributeCommonInfo.h

```diff
@@ -152,7 +152,14 @@ class AttributeCommonInfo {
     // FIXME: Eventually we want to do a list here populated via tablegen.  But
     // we want C++ attributes to be permissible on Lambdas, and get propagated
     // to the call operator declaration.
-    return getParsedKind() == AT_SYCLIntelKernelArgsRestrict;
+    // FIXME: Attribute lambda capturing is not a part of SYCL 1.2.1. We shall
```

Comment:
What do you mean by "is not a part of SYCL 1.2.1"? 
The spec is vague and doesn't explicitly mention how the attributes should be used:

> In SYCL, these attributes are legal on device functions and their specification is propagated down to any caller of those device functions

If a lambda function is considered to be a device function, then I don't see any problem with the spec.

Reply:
Ok, I agree.